exporter_test.go

```diff
@@ -602,40 +603,71 @@ type = "Apache-2.0"
 				})
 			})
 
+			when("there is project metadata", func() {
+				it("saves metadata with project info", func() {
+					fakeProjectMetadata := lifecycle.ProjectMetadata{
+						Source: lifecycle.ProjectSource{
+							Type: "git",
+							Version: map[string]interface{}{
+								"commit": "abcd1234",
+							},
+							Metadata: map[string]interface{}{
+								"repository": "github.com/buildpack/lifecycle",
+								"branch":     "master",
+							},
+						}}
+					h.AssertNil(t, exporter.Export(layersDir, appDir, fakeAppImage, runImageRef, fakeImageMetadata, additionalNames, launcherConfig, stack, fakeProjectMetadata))
+
+					projectJSON, err := fakeAppImage.Label("io.buildpacks.project")
+					h.AssertNil(t, err)
+
+					var projectMD lifecycle.ProjectMetadata
+					if err := json.Unmarshal([]byte(projectJSON), &projectMD); err != nil {
+						t.Fatalf("badly formatted metadata: %s", err)
+					}
+
+					t.Log("adds project metadata to label")
+					h.AssertEq(t, projectMD.Source.Type, "git")
```

Comment:
Just a suggestion: since you unmarshaled into a struct of the same type you passed in, you could simply assert that the two structs are equal for expediency.

Reply:
Good point, we've amended this 👍